dashboard/test/controllers/api/v1/census/census_controller_test.rb

```diff
@@ -0,0 +1,89 @@
+    assert_response 400
+    response = JSON.parse(@response.body)
+    refute response['form_version'].nil?, "form_version expected in errors: #{response}"
+  end
+
+  test 'census submission with bad school fails' do
+    post :new_submission,
+      params: {
+        form_version: 'anything',
+        nces_school_s: '00000044'
+      }
+    assert_response 400
+    response = JSON.parse(@response.body)
+    refute response['nces_school_s'].nil?, "nces_school_s expected in errors: #{response}"
+  end
+
+  test 'HoC census submission with good school succeeds' do
+    post :new_submission,
+      params: {
+        form_version: 'CensusHoc2017v3',
+        nces_school_s: '60000113717',
+        school_year: 2017,
+        submitter_email_address: "fake@email.address",
+        submitter_name: "Somebody"
+      }
+    assert_response 201
+    response = JSON.parse(@response.body)
+    refute response['census_submission_id'].nil?, "census_submission_id expected in response: #{response}"
+  end
+
+  test 'HoC census submission with negative 1 school_id succeeds' do
+    post :new_submission,
+      params: {
+        form_version: 'CensusHoc2017v3',
+        nces_school_s: '-1',
+        school_year: 2017,
+        country_s: "US",
+        submitter_email_address: "fake@email.address",
+        submitter_name: "Somebody"
+      }
+    assert_response 201
+    response = JSON.parse(@response.body)
+    refute response['census_submission_id'].nil?, "census_submission_id expected in response: #{response}"
+  end
+
+  test 'yourschool census submission with good school succeeds' do
+    post :new_submission,
+      params: {
+        form_version: 'CensusYourSchool2017v4',
+        nces_school_s: '60000113717',
+        school_year: 2017,
+        submitter_email_address: "fake@email.address",
+        submitter_name: "Somebody",
+        submitter_role: 'OTHER',
+        how_many_do_hoc: 'NONE',
+        how_many_after_school: 'NONE',
+        how_many_10_hours: 'NONE',
+        how_many_20_hours: 'NONE'
+      }
+    assert_response 201, @response.body.to_s
+    response = JSON.parse(@response.body)
+    refute response['census_submission_id'].nil?, "census_submission_id expected in response: #{response}"
+  end
+
+  test 'yourschool census submission with negative 1 school_id succeeds' do
+    post :new_submission,
+      params: {
+        form_version: 'CensusYourSchool2017v4',
+        nces_school_s: '-1',
+        school_year: 2017,
+        country_s: "US",
+        submitter_email_address: "fake@email.address",
+        submitter_name: "Somebody",
+        submitter_role: 'OTHER',
+        how_many_do_hoc: 'NONE',
+        how_many_after_school: 'NONE',
+        how_many_10_hours: 'NONE',
+        how_many_20_hours: 'NONE'
+      }
+    assert_response 201, @response.body.to_s
+    response = JSON.parse(@response.body)
+    refute response['census_submission_id'].nil?, "census_submission_id expected in response: #{response}"
+  end
```

Comment:
Is it worth testing other variations of submissions beyond school? invalid email, etc?

Reply:
Or confirming schools with punctuation succeed?

Reply:
What do you mean by "schools with punctuation"?